lib/ui/painting/image_decoding.cc

```diff
@@ -35,20 +35,20 @@ sk_sp<SkImage> DecodeImage(std::vector<uint8_t> buffer) {
   if (sk_data == nullptr)
     return nullptr;
 
-  std::unique_ptr<SkImageGenerator> generator(
-      SkImageGenerator::NewFromEncoded(sk_data.get()));
+  sk_sp<SkImage> image = SkImage::MakeFromEncoded(std::move(sk_data));
```

Comment:
We could transfer ownership of the buffer vector's data to an SkData without doing a copy here.  See https://github.com/flutter/engine/pull/3426

Alternatively, DecodeImageFromList could construct an SkData holding a copy of the Dart data and pass that instead of using a vector